### PR TITLE
config: add init fetch timeout stat

### PR DESCRIPTION
**Pull request by ramaraochavali**
Description: Adds a stat for init fetch timeout
Risk Level: Low
Testing: Added
Docs Changes: Updated
Release Notes: Added

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@mattklein123  can you PTAL?

File: docs/root/configuration/cluster_manager/cds.rst [hunk 1]
```diff
@@ -24,6 +24,7 @@ CDS has a statistics tree rooted at *cluster_manager.cds.* with the following st
   :widths: 1, 1, 2
 
   config_reload, Counter, Total API fetches that resulted in a config reload due to a different config
+  init_fetch_timeout, Counter, Total :ref:`initial fetch timeouts <envoy_api_field_core.ConfigSource.initial_fetch_timeout>`
```
Comment (mattklein123) [inline docs/root/configuration/cluster_manager/cds.rst]: Thanks, this is a really useful change. However, I think with all of our xDS, it's going to be hard to keep each doc spot up to date. WDYT about maybe have a central doc for "common subscription stats" which we can link to from all the various xDS documentation? cc @htuch

Comment (ramaraochavali) [reply docs/root/configuration/cluster_manager/cds.rst]: That makes sense. I can add a central doc for "common subscription stats" - I can do a follow-up if it is easier to review the just doc part or if you think it is better to do it as part of this, I am fine with that too. LMK what you think?

Comment (mattklein123) [reply docs/root/configuration/cluster_manager/cds.rst]: I think it's fine to do it as part of this change as it's a pretty small change. Thank you!

Comment (htuch) [reply docs/root/configuration/cluster_manager/cds.rst]: Yep, good idea!

Comment (ramaraochavali) [reply docs/root/configuration/cluster_manager/cds.rst]: @mattklein123 added the subscription_stats page and linked every where else. PTAL.

File: docs/root/configuration/subscription_stats.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
+.. _subscription_statistics:
+
+Subscription statistics
+=======================
+
+The following statistics are generated for all subscriptions.
```
Comment (mattklein123) [inline docs/root/configuration/subscription_stats.rst]: Can you clarify what a "subscription" is and maybe cross link to relevant xDS/APIv2 docs? Maybe the title of this should be xDS subscription? WDYT? (Basically I think to the casual reader it won't be clear what this page is about)

Comment (ramaraochavali) [reply docs/root/configuration/subscription_stats.rst]: Sure. Will do.

**Comment by ramaraochavali** [comment]
@mattklein123 changed to xds subscription, clarified what subscription with cross reference. PTAL.
Regarding, docs on a bunch of other xDS APIs , yeah it seems we are missing these stats docs for many xDS apis.

**Pull request merged**